packages/wonder-blocks-core/util/add-style.md

```diff
@@ -164,4 +172,100 @@ const styles = StyleSheet.create({
 });
 
 <DynamicStyles />
-```
\ No newline at end of file
+```
+
+#### 4. Using the `style` prop
+
+There are some cases where you don't need to use `addStyle`. For example, if you
+need to create a custom component that is using a Wonder Blocks component, you
+can create a `style` prop and pass it down to the WB component itself.
+
+**NOTE:** Make sure to import the `StyleType` type definition and assign it to
+the prop:
+
+```js static
+import type {StyleType} from "@khanacademy/wonder-blocks-core";
+
+type Props = {
+    style: StyleType,
+    // add the other props here
+};
+```
+
+```jsx
+import {StyleSheet} from "aphrodite";
+import Color from "@khanacademy/wonder-blocks-color";
+import {Checkbox} from "@khanacademy/wonder-blocks-form";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import {Title} from "@khanacademy/wonder-blocks-typography";
+import {addStyle, View} from "@khanacademy/wonder-blocks-core";
+
+// you'll need to import the type definition here
+// import type {StyleType} from "@khanacademy/wonder-blocks-core";
+
+class CustomComponent extends React.Component {
+    render() {
+        return (
+            <View style={[
+                styles.default,
+                // this `style` prop should be of type `StyleType`
+                this.props.style,
+            ]}>
+                <Title>Lorem ipsum</Title>
+            </View>
+        );
+    }
+}
+
+const styles = StyleSheet.create({
+    // default styles
+    default: {
+        background: Color.lightBlue,
+        color: Color.white,
+        padding: Spacing.xxLarge,
+    },
+    // style to be passed as a prop
+    customStyle: {
+        background: Color.darkBlue,
+    },
+});
+
+class UsingStyleExample extends React.Component {
```

Comment:
This example feels overly complex.  I'd drop `UsingStyleExample` and just do:
```
<CustomComponent style={styles.customStyle} />
```

Reply:
I initially created the example that way, but as I also wanted to illustrate how the styles are injected in runtime, I decided to add that extra part. If you still feel it's too complex, I definitively can drop that extra component.

Reply:
re🤔 about this, you're right, it feels more complex than it needs to be. I'm going to drop that extra component.